### PR TITLE
lab 10

### PR DESCRIPTION
**Pull request by amyslytska**

### CODE REVIEW

File: lecture_notes/10 refactoring/1-not-optimal.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import ast
+def shift(offset, points):
```
Comment (tshemsedinov) [inline lecture_notes/10 refactoring/1-not-optimal.py:2]: Add single empty line to separate function

File: lecture_notes/10 refactoring/2-style-fix.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+'use strict';
+const shift = (offset, points) => {
```
Comment (tshemsedinov) [inline lecture_notes/10 refactoring/2-style-fix.js:2]: +1 empty line

File: lecture_notes/10 refactoring/7-optimized.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import ast
+
+def shift(offset, points):
+    for point in points:
+        point['x'] += offset['x']
+        point['y'] += offset['y']
+    return points
+
+def parsing(points):
+    parsed = [ast.literal_eval(point) if type(point) == str else point for point in points]
```
Comment (tshemsedinov) [inline lecture_notes/10 refactoring/7-optimized.py:10]: Very long expression, optimize for readability

File: lecture_notes/10 refactoring/7-optimized.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import ast
+
+def shift(offset, points):
+    for point in points:
+        point['x'] += offset['x']
+        point['y'] += offset['y']
+    return points
```
Comment (tshemsedinov) [inline lecture_notes/10 refactoring/7-optimized.py:7]: Do not change incoming parameters, return new list

File: lecture_notes/10 refactoring/7-optimized.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
+'use strict';
+
+const shift = (offset) => (point) => {
+  const shifted = {
+    x: (point.x += offset.x),
+    y: (point.y += offset.y),
```
Comment (tshemsedinov) [inline lecture_notes/10 refactoring/7-optimized.js:6]: Why do you use `+=` operator here?

**Review by tshemsedinov — CHANGES_REQUESTED** [review]
- Don not modify incoming structures like list and point, create new one instead
- Fix js version

File: lecture_notes/10 refactoring/7-optimized.js (hunk 5 above)
Comment (amyslytska) [reply lecture_notes/10 refactoring/7-optimized.js]: I fixed everything you mentioned, can I please get some more points now?

**Pull request closed without merging**